src/compile/legend/assemble.ts

```diff
@@ -26,13 +26,18 @@ export function assembleLegends(model: Model): VgLegend[] {
 
   return flatten(vals(legendByDomain)).map((legendCmpt: LegendComponent) => {
     const legend = legendCmpt.combine();
-    // For non color channel's legend, we need to override symbol stroke config from Vega config
+
     if (legend.encode && legend.encode.symbols) {
       const out = legend.encode.symbols.update;
       if (out.fill && out.fill['value'] !== 'transparent' && !out.stroke && !legend.stroke) {
         // For non color channel's legend, we need to override symbol stroke config from Vega config if stroke channel is not used.
         out.stroke = {value: 'transparent'};
       }
+
+      if (legend.fill && legend.size) {
+        // If fill and size are both used, for non color channel's legend, we need remove fill, as it is already specified.
```

Comment:
Shouldn't we always delete this when fill is defined?

Reply:
Wasn't sure if under other conditions you may want an override, but I guess that does not make much sense. Fixed.